[clang-tidy] Fix false positives about references in `misc-const-correctness` (#160971)
llvm/llvm-project: clang-tools-extra/docs/ReleaseNotes.rst

  - ``for`` loops are supported.

- Improved :doc:`misc-const-correctness
  <clang-tidy/checks/misc/const-correctness>` check to avoid false

>   positives when pointers is tranferred to non-const references.

Reviewer: ```suggestion
  positives when pointers to non-const are bound to references.
```
Since the false positive happens [whether the reference is const or not](https://godbolt.org/z/5zGvrW39n).
